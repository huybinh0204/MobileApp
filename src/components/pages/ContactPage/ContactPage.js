import React from 'react';
import { View } from 'react-native';
import { IconSvg, Link } from '_components/atoms';
import { SecondaryScreenLayout } from '_components/organisms';
import { ICONS, LINK_TYPES } from '_constants';
import strings from '_localization';
import { normalize } from '_utilities/screen';
import {
  BottomContainer,
  ContactButton,
  Content,
  LinkText,
  MainTitle,
  Note,
  RegularText,
  SupportContainer,
  SupportTitle,
} from './ContactPage.styles';

const ContactPage = () => {
  return (
    <SecondaryScreenLayout testID="ContactPage" title={strings.settings.contact}>
      <Content>
        <IconSvg icon={ICONS.contact} width={normalize(270)} height={normalize(270)} />
        <MainTitle>{strings.contact.main_title}</MainTitle>
        <SupportContainer>
          <RegularText>
            {strings.contact.contactSupport}
            <Link href={strings.contact.email} type={LINK_TYPES.MAIL} wrapperComponent={LinkText}>
              {strings.contact.email}
            </Link>
            {strings.contact.forAssistance}
          </RegularText>
        </SupportContainer>
        <View>
          <SupportTitle>{strings.contact.supportTitle}</SupportTitle>
          <RegularText>{strings.contact.supportSchedule}</RegularText>
        </View>
        <BottomContainer>
          <Note>{strings.contact.supportAvailability}</Note>
          <Link
            accessibilityLabel="ContactButton"
            href={strings.kinly_support_number}
            type={LINK_TYPES.PHONE}
            wrapperComponent={ContactButton}
          >
            {strings.contact.button}
          </Link>
        </BottomContainer>
      </Content>
    </SecondaryScreenLayout>
  );
};

export default ContactPage;
