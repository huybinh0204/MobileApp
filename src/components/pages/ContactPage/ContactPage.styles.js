import styled from '@emotion/native';
import { BaseText, MainButton } from '_components/atoms';

export const Content = styled.View(({ theme }) => ({
  flex: 1,
  alignItems: 'center',
  justifyContent: 'flex-end',
  padding: theme.spacing.m,
}));

export const MainTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading4,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));

export const RegularText = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta900,
  textAlign: 'center',
}));

export const ContactButton = styled(MainButton)(({ theme }) => ({
  marginTop: theme.spacing.s,
}));

export const LinkText = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.alpha500,
}));

export const SupportTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentSemiBold,
  color: theme.colors.beta500,
}));

export const Note = styled(BaseText)(({ theme }) => ({
  ...theme.typography.legal,
  color: theme.colors.beta500,
}));

export const SupportContainer = styled.View(({ theme }) => ({
  marginBottom: theme.spacing.s,
}));

export const BottomContainer = styled.View({
  flex: 1,
  width: '100%',
  alignItems: 'center',
  justifyContent: 'flex-end',
});
