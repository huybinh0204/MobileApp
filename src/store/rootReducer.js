import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
