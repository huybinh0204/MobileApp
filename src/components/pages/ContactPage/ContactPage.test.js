import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Linking } from 'react-native';
import { createStore } from 'redux';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import ContactPage from './ContactPage';
import strings from '_localization';

describe('ContactPage', () => {
  let mockStore;

  beforeEach(() => {
    mockStore = createStore(rootReducer, {});
    jest.spyOn(Linking, 'openURL');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<ComponentWithProviders component={ContactPage} store={mockStore} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should show the contact button', async () => {
    const { getByA11yLabel } = render(
      <ComponentWithProviders component={ContactPage} store={mockStore} />
    );

    await fireEvent.press(getByA11yLabel('ContactButton'));
    expect(Linking.openURL).toBeCalled();
  });

  it('should render the contact email as email link', async () => {
    const { getByText } = render(
      <ComponentWithProviders component={ContactPage} store={mockStore} />
    );

    await fireEvent.press(getByText(strings.contact.email));
    expect(Linking.openURL).toBeCalledWith(`mailto:${strings.contact.email}`);
  });
});
