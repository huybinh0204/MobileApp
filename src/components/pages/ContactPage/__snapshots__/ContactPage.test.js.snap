// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="ContactPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#1F1D25",
                                  "fontFamily": "Poppins-SemiBold",
                                  "fontSize": 29,
                                  "lineHeight": 40,
                                },
                                undefined,
                              ]
                            }
                          >
                            Contact
                          </Text>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "justifyContent": "flex-end",
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <SvgMock
                              height={489}
                              width={489}
                            />
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 43.5,
                                    "lineHeight": 60,
                                    "marginBottom": 29,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Have a question?
                            </Text>
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginBottom": 29,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Contact support at 1-888-795-4659 or

                                <Text
                                  allowFontScaling={false}
                                  onPress={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#3D3397",
                                        "fontFamily": "Poppins",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                      },
                                      undefined,
                                    ]
                                  }
                                  wrapperProps={null}
                                >
                                  support@bekinly.com
                                </Text>
                                
for assistance.
                              </Text>
                            </View>
                            <View>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#5F6A77",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Customer Support Hours*
                              </Text>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Mon-Sat 7am-7pm CST
 Sun 9am-5pm CST
                              </Text>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flex": 1,
                                    "justifyContent": "flex-end",
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#5F6A77",
                                      "fontFamily": "Poppins",
                                      "fontSize": 20,
                                      "lineHeight": 28,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                *Holidays excluded
                              </Text>
                              <View
                                accessibilityLabel="ContactButton"
                                accessibilityState={
                                  Object {
                                    "disabled": false,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "#3D3397",
                                    "borderRadius": 10,
                                    "marginTop": 29,
                                    "opacity": 1,
                                    "paddingHorizontal": 45.5,
                                    "paddingVertical": 31,
                                  }
                                }
                              >
                                <Text
                                  allowFontScaling={false}
                                  disabled={false}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "textAlign": "center",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  Call us
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
